transport_analysis/conductivity_integration.py

```diff
@@ -0,0 +1,342 @@
+        as explained here: https://stackoverflow.com/questions/34222272/computing-mean-square-displacement-using-python-and-fft
+        Since we are using FFT, the sampling frequency (or) dump frequency of the simulation trajectory
+        plays a critical role in the accuracy of the result.
+
+        Particle velocities are required to calculate the viscosity function. Thus
+        you are limited to MDA trajectories that contain velocity information, e.g.
+        GROMACS .trr files, H5MD files, etc. See the MDAnalysis documentation:
+        https://docs.mdanalysis.org/stable/documentation_pages/coordinates/init.html#writers.
+
+        Parameters
+        ----------
+    atomgroup : AtomGroup
+        An MDAnalysis :class:`~MDAnalysis.core.groups.AtomGroup`.
+        Note that :class:`UpdatingAtomGroup` instances are not accepted.
+    temp_avg : float (optional, default 300)
+        Average temperature over the course of the simulation, in Kelvin.
+    dim_type : {'xyz', 'xy', 'yz', 'xz', 'x', 'y', 'z'}
+        Desired dimensions to be included in the viscosity calculation.
+        Defaults to 'xyz'.
+    linear_fit_window : tuple of int (optional)
+        A tuple of two integers specifying the start and end lag-time for
+        the linear fit of the viscosity function. If not provided, the
+        linear fit is not performed and viscosity is not calculated.
+
+    Attributes
+    ----------
+    atomgroup : :class:`~MDAnalysis.core.groups.AtomGroup`
+        The atoms to which this analysis is applied
+    dim_fac : int
+        Dimensionality :math:`d` of the viscosity computation.
+    results.timeseries : :class:`numpy.ndarray`
+        The averaged viscosity function over all the particles with respect
+        to lag-time. Obtained after calling :meth:`ViscosityHelfand.run`
+    results.visc_by_particle : :class:`numpy.ndarray`
+        The viscosity function of each individual particle with respect to
+        lag-time.
+    results.viscosity : float
+        The viscosity coefficient of the solvent. The argument
+        `linear_fit_window` must be provided to calculate this to
+        avoid misinterpretation of the viscosity function.
+    start : Optional[int]
+        The first frame of the trajectory used to compute the analysis.
+    stop : Optional[int]
+        The frame to stop at for the analysis.
+    step : Optional[int]
+        Number of frames to skip between each analyzed frame.
+    n_frames : int
+        Number of frames analysed in the trajectory.
+    n_particles : int
+        Number of particles viscosity was calculated over.
+    """
+
+    def __init__(
+        self,
+        allatomgroup: "AtomGroup",
+        cationgroup_query: str,
+        aniongroup_query: str,
+        temp_avg: float = 298.0,
+        cutoff_step: int = 10000,
+        cation_num_atoms_per_species=int,
+        anion_num_atoms_per_species=int,
+        cation_mass_per_species=float,
+        anion_mass_per_species=float,
```

Comment:
`X_num_atoms_per_species` can be calculated internally and masses are available at `atoms.masses`, no need to include any of these as `kwargs`

```python

cations = # define cation atom group
cation_num_atoms_per_species = int(len(cations) / len(cations.residues))
# same for anions
cation_mass_per_species = 

```